### ISSUE TITLE
Add PyTorch-based numerical correctness tests for tensor operators

### ISSUE THREAD

**Issue by npatsakula** [npatsakula/svod#29]
## Summary

Add comprehensive numerical correctness tests comparing Morok tensor operators against PyTorch reference implementations.

## Background

Currently, tensor operator tests verify shape and dtype correctness but don't validate numerical accuracy. Tinygrad's test suite compares outputs against PyTorch for numerical validation.

## Operators to Test

High priority (recently added):
- [ ] `batchnorm` - Compare with `torch.nn.functional.batch_norm`
- [ ] `cat` - Compare with `torch.cat`
- [ ] `try_pad` - Compare with `torch.nn.functional.pad`
- [ ] `shape_tensor` - Compare with `torch.tensor(x.shape)`

Existing operators that need numerical validation:
- [ ] Activation functions: `relu`, `sigmoid`, `tanh`, `softmax`, `gelu`, `silu`
- [ ] Reduction ops: `sum`, `mean`, `var`, `std`, `max`, `min`
- [ ] Shape ops: `reshape`, `permute`, `transpose`, `expand`, `squeeze`, `unsqueeze`

### CI/CD
- [ ] Add PyTorch to CI environment (or use separate test job)
- [ ] Configure test feature flag for torch-dependent tests (e.g., `#[cfg(feature = "torch_tests")]`)

### NIX
- [ ] Add `python3Packages.torch` to dev shell
- [ ] Create separate flake output for torch-enabled tests

## Test Structure

```rust
#[cfg(feature = "torch_tests")]
mod torch_comparison {
    use tch::Tensor as TorchTensor;
    
    #[test]
    fn test_batchnorm_vs_torch() {
        // Create input in both frameworks
        // Compare outputs within tolerance
    }
}
```

## References

- Tinygrad's approach: `test/backend/test_nn.py` compares against PyTorch
- `tch-rs` crate for Rust PyTorch bindings: https://github.com/LaurentMazare/tch-rs

## Acceptance Criteria

- All tensor operators have numerical comparison tests
- Tests run in CI with appropriate feature flag
- Tolerance values documented for float comparisons
